actions/dataset.go

```diff
@@ -77,24 +90,26 @@ func SaveDataset(node *p2p.QriNode, changes *dataset.Dataset, secrets map[string
 		}
 	}
 
-	// apply the changes to the previous dataset.
-	mutable.Assign(changes)
-	changes = mutable
+	if !sw.Replace {
+		// apply the changes to the previous dataset.
```

Comment:
I like using the language "patch" to describe the non-replace behavior. Would put that here for future reference, by changing the comment to "Treat the changes as a set of patches applied to the previous dataset"